azerothcore/azerothcore-wotlk
# fix(Core/DB): Fire Elemental Invasion changes

## Changes Proposed:
- Baron Charr added to Elemental Invasion game event (13)
- Blazing Invader's SAI rework.
- SMART_ACTION_SET_COUNTER extra parameter for subtracting values to the counter.
- Changed `Map::TempSummon* SummonCreature` to have WorldObjects as summoner and not just Units.
- Added a nullptr check for `SmartScript::GetTargets`

## Issues Addressed:

- Closes #8080 
https://github.com/chromiecraft/chromiecraft/issues/1440

## SOURCE:


## Tests Performed:

- Compiled and tested in Windows 10


## How to Test the Changes:


1. Go to ungoro and check the elemental Invasion
2. Only maximum of 3 Blazings should spawn at a time per Rift
3. After one is killed, after a while another spawns.

## Known Issues and TODO List:


None


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
